code/game/objects/items/cigs_lighters.dm

```diff
@@ -297,19 +297,33 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		qdel(src)
 	. = ..()
 
-/obj/item/clothing/mask/cigarette/afterattack(mob/M, mob/user, proximity)
+/obj/item/clothing/mask/cigarette/afterattack(mob/living/M, mob/living/user, proximity)
 	if(!istype(M))
 		return ..()
+	if(HAS_TRAIT(user, TRAIT_PACIFISM))
```

Comment:
I cant test it right now, but is this check for the pacifism trait needed? It should already stop the attack chain during the attack() proc right?

Reply:
Since I added extra damage for butting it out on someone it would bypass the pacifist trait and add the 4 extra damage anyway, it would also trigger the code to tell it to remove smoke time and extinguish the cigarette which i thought didn't make sense if you're not hurting anyone.